dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/acl/TrackedEntityInstanceAclReadTests.java

```diff
@@ -0,0 +1,539 @@
+
+        private boolean allAuthority;
+
+        public User( String username, String uid, String password )
+        {
+            this.username = username;
+            this.uid = uid;
+            this.password = password;
+        }
+
+        public String getUsername()
+        {
+            return username;
+        }
+
+        public void setUsername( String username )
+        {
+            this.username = username;
+        }
+
+        public String getUid()
+        {
+            return uid;
+        }
+
+        public void setUid( String uid )
+        {
+            this.uid = uid;
+        }
+
+        public String getPassword()
+        {
+            return password;
+        }
+
+        public void setPassword( String password )
+        {
+            this.password = password;
+        }
+
+        public Map<String, List<String>> getDataRead()
+        {
+            return dataRead;
+        }
+
+        public void setDataRead( Map<String, List<String>> dataRead )
+        {
+            this.dataRead = dataRead;
+        }
+
+        public Map<String, List<String>> getNoDataRead()
+        {
+            return noDataRead;
+        }
+
+        public void setNoDataRead( Map<String, List<String>> noDataRead )
+        {
+            this.noDataRead = noDataRead;
+        }
+
+        public List<String> getGroups()
+        {
+            return groups;
+        }
+
+        public void setGroups( List<String> groups )
+        {
+            this.groups = groups;
+        }
+
+        public List<String> getCaptureScope()
+        {
+            return captureScope;
+        }
+
+        public void setCaptureScope( List<String> captureScope )
+        {
+            this.captureScope = captureScope;
+        }
+
+        public List<String> getSearchScope()
+        {
+            return searchScope;
+        }
+
+        public void setSearchScope( List<String> searchScope )
+        {
+            this.searchScope = searchScope;
+        }
+
+        public List<String> getScopes()
+        {
+            return ListUtils.union( searchScope, captureScope );
+        }
+
+        public void setAllAuthority( boolean allAuthority )
+        {
+            this.allAuthority = allAuthority;
+        }
+
+        public boolean getAllAuthority()
+        {
+            return allAuthority;
+        }
+
+        public boolean hasAllAuthority()
+        {
+            return allAuthority;
+        }
+    }
+
+    @BeforeAll
+    public void before()
+        throws Exception
+    {
+        teiActions = new TEIActions();
+        metadataActions = new MetadataActions();
+        userActions = new UserActions();
+
+        // Setup as SuperUser
+        new LoginActions().loginAsDefaultUser();
+
+        // Set up metadata (Import twice to connect all references)
+        metadataActions.importMetadata( new File( "src/test/resources/tracker/acl/metadata.json" ) );
```

Comment:
Did you try importAndValidateMetadata() ? We might want to have validation here, since it will affect test execution if there are conflicts and metadata is not imported.